crowbar_framework/config/routes.rb

```diff
@@ -24,104 +24,105 @@
+  put "crowbar/:controller/1.0/proposals(.:format)", action: "proposal_create", as: "create_proposal_barclamp"
+  get "crowbar/:controller/1.0/proposals(.:format)", action: "proposals", as: "proposals_barclamp"
+  get "crowbar/:controller/1.0/proposals/template(.:format)", action: "proposal_template", as: "template_proposal_barclamp"
+  post "crowbar/:controller/1.0/proposals/commit/:id(.:format)", action: "proposal_commit", as: "commit_proposal_barclamp"
   get "crowbar/:controller/1.0/proposals/status(/:id)(/:name)(.:format)", action: "proposal_status", as: "status_proposals_barclamp"
-  delete "crowbar/:controller/1.0/proposals/:id", action: "proposal_delete", as: "delete_proposal_barclamp"
-  delete "crowbar/:controller/1.0/proposals/dequeue/:id", action: "proposal_dequeue", as: "dequeue_barclamp"
-  post "crowbar/:controller/1.0/proposals/:id", action: "proposal_update", as: "update_proposal_barclamp"
-  get "crowbar/:controller/1.0/proposals/:id", action: "proposal_show", as: "proposal_barclamp"
+  delete "crowbar/:controller/1.0/proposals/:id(.:format)", action: "proposal_delete", as: "delete_proposal_barclamp"
+  delete "crowbar/:controller/1.0/proposals/dequeue/:id(.:format)", action: "proposal_dequeue", as: "dequeue_barclamp"
+  post "crowbar/:controller/1.0/proposals/:id(.:format)", action: "proposal_update", as: "update_proposal_barclamp"
+  get "crowbar/:controller/1.0/proposals/:id(.:format)", action: "proposal_show", as: "proposal_barclamp"
 
-  get "crowbar/:controller/1.0/elements", action: "elements"
-  get "crowbar/:controller/1.0/elements/:id", action: "element_info"
-  post "crowbar/:controller/1.0/transition/:id", action: "transition"
-  get "crowbar/:controller/1.0/transition/:id", action: "transition"
+  get "crowbar/:controller/1.0/elements(.:format)", action: "elements"
+  get "crowbar/:controller/1.0/elements/:id(.:format)", action: "element_info"
+  post "crowbar/:controller/1.0/transition/:id(.:format)", action: "transition"
+  get "crowbar/:controller/1.0/transition/:id(.:format)", action: "transition"
 
-  get "crowbar/:controller/1.0", action: "index", as: "index_barclamp"
-  delete "crowbar/:controller/1.0/:id", action: "delete", constraints: { id: /[^\/]+/ }, as: "delete_barclamp"
-  get "crowbar/:controller/1.0/:id", action: "show", constraints: { id: /[^\/]+/ }, as: "show_barclamp"
-  get "crowbar/:controller", action: "versions", as: "versions_barclamp"
-  post "crowbar/:controller/1.0/:action/:id", constraints: { id: /[^\/]+/ }, as: "action_barclamp"
-  get "crowbar", controller: "barclamp", action: "barclamp_index", as: "barclamp_index_barclamp"
-  get "crowbar/modules/1.0", controller: "barclamp", action: "modules", as: "barclamp_modules"
+  get "crowbar/:controller/1.0(.:format)", action: "index", as: "index_barclamp"
+  delete "crowbar/:controller/1.0/:id(.:format)", action: "delete", constraints: { id: /[^\/]+/ }, as: "delete_barclamp"
```

Comment:
Metrics/LineLength: Line is too long. [120/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)